src/Network/Probe.php

```diff
@@ -440,7 +440,7 @@ public static function uri($uri, $network = '', $uid = -1, $cache = true)
 			}
 		}
 
-		if (!empty(self::$baseurl)) {
+		if (empty($data['baseurl']) && !empty(self::$baseurl)) {
```

Comment:
The `baseurl` in `self::$baseurl` is the most reliable one, since it it the base path we detected during the probing. So this should always have priority when fetching.

Reply:
See my other change in the `twitter()` method, `self::$baseurl` isn't reliable. With the added hook, there are even less chances for this variable to be set.

Reply:
The `self::$baseurl` is set with the `host-meta` fetching. So this path is completely reliable. The `baseurl` field is also set via some provided data - thus less reliable.

Reply:
Possibly you could provide the `self::$baseurl` as a parameter in the hook as well?

Reply:
The problem is that there aren't any guarantee `host-meta` was called for the current profile URL detection. Maybe the `baseurl` property shouldn't be set in `Probe::detect:770-775` if it isn't reliable?

Reply:
Yes. That code is crap.

Reply:
Done.